src/Http/Request/Body/JsonSchemaValidatingRequestBodyParser.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Http\Request\Body;
+
+use CultuurNet\UDB3\Http\ApiProblem\ApiProblem;
+use CultuurNet\UDB3\Http\ApiProblem\SchemaError;
+use Opis\JsonSchema\Errors\ErrorFormatter;
+use Opis\JsonSchema\Validator;
+use Psr\Http\Message\ServerRequestInterface;
+
+final class JsonSchemaValidatingRequestBodyParser implements RequestBodyParser
+{
+    private const SCHEMA_LOCATIONS = __DIR__ . '/../../../../vendor/publiq/stoplight-docs-uitdatabank/models/';
+    private const MAX_ERRORS = 100;
+
+    private RequestBodyParser $baseParser;
+    private Validator $validator;
+    private string $jsonSchema;
+
+    private function __construct(string $jsonSchema)
+    {
+        $this->jsonSchema = $jsonSchema;
+        $this->baseParser = new JsonRequestBodyParser();
```

Comment:
This will need to become configurable in a follow-up PR to make it possible to add use a custom `RequestBodyParser` that first transforms the data before validating it. But since that is not needed for the endpoints that use it now (`PUT /events/{eventId}/bookingAvailability` and `PATCH /events/{eventId}/subEvents`) I decided to hardcode it to `JsonRequestBodyParser` for now here.